### DISCUSSION TITLE
Pulse Counter GPIO

### DISCUSSION THREAD

**Discussion by jtricerolph** [jomjol/AI-on-the-edge-device#1778] [Ideas]
Havnt seen this mentioned but a pulse counter option would be cool.
My meter limits me to whole KWh leaving very jumpy rate values, looking to swap it out for electrical/digital one with pulse led. With 1000imp/kwh this resolution should give a reasonable rate calculation potentially on a more frequent update rate to give a more fluid rate display more like DNO style smart meter monitors or CT clamp style?
Just a thought.

**Comment by De-Backer** [comment]
hello everyone,
I would like to write this in the code, the Pulse Counter is not that difficult. 
But, what are the requirements that must be met?
- The pin must of course be freely selectable.
- Link pulse to take photo?
- Measure the time between pulses?
- And then there are decoding standards, in a separate thread? Separate file's? .....
Please some guidelines.

**Reply by caco3** [reply]
Or project is focussed on non-digital meters with pointers or digits.

A pulse counter with pulsing LEDs is a completely different concept and not covered by our project.
i think it would be easier to find a different project or start one from scratch.

**Reply by W0w3** [reply]
How about adding a reed switch to one of the GPIO ports?
This could then trigger:
1. a MQTT Message
2. a new recognition cycle after the switch did not change for x seconds

I have found I can place a xiaomi window contact into my gas meter and it will get triggered everytime the last digit has turned to 0.
Unfurtunately the zigbee connection to that sensor is unreliable. It would be nice to attach that reed switch as an additional sensor to the already available and reliable ESP-32.

**Reply by caco3** [reply]
The project is opensource, you you can extend it to your needs, but we will not add this feature by ourself.
